scrapy/utils/sitemap.py

```diff
@@ -34,10 +37,13 @@ def __iter__(self):
                 yield d
 
 
-def sitemap_urls_from_robots(robots_text):
+def sitemap_urls_from_robots(robots_text, base_url=None):
     """Return an iterator over all sitemap urls contained in the given
     robots.txt file
     """
     for line in robots_text.splitlines():
         if line.lstrip().lower().startswith('sitemap:'):
-            yield line.split(':', 1)[1].strip()
+            url = line.split(':', 1)[1].strip()
+            if not re.match('https?://', url, re.I):
```

Comment:
I think case doesn't matter here, and converting protocol to lower case is fine. A good catch though!